framework/web/Request.php

```diff
@@ -162,7 +162,39 @@ class Request extends \yii\base\Request
      * @see getBodyParams()
      */
     public $parsers = [];
+    /**
+     * @var array The configuration for trusting security related header.
+     * The array key is a regular expression for matching the hostname.
+     * The value is a list of headers to trust.
+     * Example 1: trust all headers from domains ending in '.trusted.com'
+     * Example 2: trust just the x-forwarded-for header from domains ending in '.partial.com'
+     * ```
+     * [
+     *     '/^.*\.trusted.com$/',
+     *     '/^.*\.partial.com$/' => ['X-Forwarded-For']
+     *
+     * ]
+     * ```
+     * The default value trusts all headers from all hosts (to not break BC).
+     */
+    public $trustedHostConfig = [
+        '//'
+    ];
```

Comment:
> That's possible as well, I'm just really against inventing new pattern types

We already using `fnmatch` for matching hosts and ips in some places, using regexp here is inconsistent. 

> Every decent developer knows how to use regular expressions and they are more powerful.

And they are tricky and less readable. Even in your example there is not escaped dot. :)